src/openrct2/world/Footpath.cpp

```diff
@@ -1049,34 +1045,32 @@ void footpath_connect_edges(int32_t x, int32_t y, TileElement* tileElement, int3
  *  rct2: 0x006A742F
  */
 void footpath_chain_ride_queue(
-    ride_id_t rideIndex, int32_t entranceIndex, int32_t x, int32_t y, TileElement* tileElement, int32_t direction)
+    ride_id_t rideIndex, int32_t entranceIndex, const CoordsXY& footpathPos, TileElement* tileElement, int32_t direction)
 {
     TileElement *lastPathElement, *lastQueuePathElement;
-    int32_t lastPathX = x, lastPathY = y, lastPathDirection = direction;
+    auto lastPath = CoordsXYZD{ footpathPos, tileElement->GetBaseZ(), static_cast<Direction>(direction) };
```

Comment:
Create a new variable called `curQueuePos` initial value of `initialFootpathPos`